lib/codecs/src/decoding/format/json_paths.rs

```diff
@@ -0,0 +1,1182 @@
+use actson::feeder::PushJsonFeeder;
+use actson::options::JsonParserOptionsBuilder;
+use actson::{JsonEvent, JsonParser};
+use bytes::Bytes;
+use derivative::Derivative;
+use lookup::OwnedValuePath;
+use smallvec::{smallvec, SmallVec};
+use std::collections::{BTreeMap, HashMap};
+use vector_config::configurable_component;
+use vector_core::{
+    config::{DataType, LogNamespace},
+    event::{Event, LogEvent},
+    schema,
+};
+use vrl::value::{Kind, Value};
+
+use super::{default_lossy, Deserializer};
+
+/// Operations that can be performed on JSON paths
+#[configurable_component]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[serde(rename_all = "lowercase")]
+pub enum PathOperation {
+    /// Emit the value as-is when encountered
+    Identity,
+    /// Emit each array element as a separate event
+    Explode,
+    /// Emit the value as bytes
+    Bytes,
+}
+
+
+/// Configuration for path-based operations
+#[configurable_component]
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct PathOperationConfig {
+    /// Map of JSON paths (as strings) to operations to perform on them
+    pub paths: BTreeMap<String, PathOperation>,
+}
+
+impl PathOperationConfig {
+    /// Create a new configuration with at least one path
+    pub fn new(first_path: OwnedValuePath, first_op: PathOperation) -> Self {
+        let mut paths = BTreeMap::new();
+        paths.insert(first_path.to_string(), first_op);
+        Self { paths }
+    }
+
+    /// Add a path to the configuration (builder pattern)
+    pub fn with_path(mut self, path: OwnedValuePath, operation: PathOperation) -> Self {
+        self.paths.insert(path.to_string(), operation);
+        self
+    }
+
+    /// Create from HashMap (fallible, for compatibility)
+    pub fn from_map(paths: HashMap<OwnedValuePath, PathOperation>) -> Result<Self, String> {
```

Comment:
`impl From<Map<Into<OwnedValuePath>, PathOperation>> for PathOpConfig` may be nicer?